plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -87,4 +99,48 @@ private Properties getCredentialProperties(ConnectorIdentity identity)
         properties.putAll(credentialPropertiesProvider.getCredentialProperties(identity));
         return properties;
     }
+
+    public static Builder builder(Driver driver, String connectionUrl, CredentialProvider credentialProvider)
+    {
+        return new Builder(driver, connectionUrl, credentialProvider);
+    }
+
+    public static class Builder
+    {
+        private final Driver driver;
+        private final String connectionUrl;
+        private Properties connectionProperties = new Properties();
+        private CredentialPropertiesProvider<String, String> credentialPropertiesProvider;
```

Comment:
can we have a "throwing" provider here?

Reply:
I'm not sure how we can use it. Could you elaborate on that?